### ISSUE TITLE
TypeScript Config (tsconfig.json) Not being Honored

### ISSUE THREAD

**Issue by arietrouw** [depcheck/depcheck#770]
### Bug Description

When specifying additional types in the compilerOptions>types section of a tsconfig.json, those packages are still reported as unused

Where the dependency is used:

```
{
  "compilerOptions": {
    "declaration": true,
    "declarationMap": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "incremental": true,
    "noImplicitOverride": false,
    "typeRoots": ["./node_modules/@types", "**/src/types"],
    "types": [
      "@types/jest",
      "jest-extended",
      "jest-sorted",
      "jest",
      "node",
      "reflect-metadata"
    ]
  },
  "exclude": ["**/dist", "**/node_modules"],
  "extends": "@xylabs/tsconfig"
}
```

Where the dependency is listed in package.json:

```
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.20.2",
    "@types/node": "^18.11.18",
    "@xylabs/eslint-config": "^2.12.8",
    "@xylabs/ts-scripts-yarn3": "^2.12.8",
    "@xylabs/tsconfig": "^2.12.8",
    "babel-jest": "^29.3.1",
    "dotenv": "^16.0.3",
    "eslint": "^8.32.0",
    "jest": "^29.3.1",
    "jest-environment-jsdom": "^29.3.1",
    "jest-extended": "^3.2.3",
    "jest-sorted": "^1.0.14",
    "nodemon": "^2.0.20",
    "ts-jest": "^29.0.5",
    "ts-node": "^10.9.1",
    "typedoc": "^0.23.24",
    "typescript": "^4.9.4"
  },
```

### Versions

- `node -v`: 16.18.1
- `npm -v`:  8.19.2
- `depcheck --version`: 1.4.3
- 'yarn': 3.3.1

### Extra info

The reason for adding these types is so that in jest files the types can be resolved.

This can be seen by pulling this repo: https://github.com/XYOracleNetwork/sdk-xyo-client-js

You would have to remove the files from the .depcheckrc file to exhibit the problem

To run depcheck in this mono-repo use: 'yarn xy deps' after a 'yarn install'

BONUS POINTS: If you wish to help us debug further, the `depcheck --json` option will show more information. Check [here in the README](https://github.com/depcheck/depcheck#example) for documentation on json output.

{"dependencies":[],"devDependencies":["@xylabs/ts-scripts-yarn3","depcheck"],"missing":{},"using":{"@xylabs/eslint-config":["/Users/arietrouw/GitHub/XYOracleNetwork/sdk-xyo-client-js/.eslintrc"],"eslint":["/Users/arietrouw/GitHub/XYOracleNetwork/sdk-xyo-client-js/.eslintrc"],"nodemon":["/Users/arietrouw/GitHub/XYOracleNetwork/sdk-xyo-client-js/package.json"],"@xyo-network/modules":["/Users/arietrouw/GitHub/XYOracleNetwork/sdk-xyo-client-js/src/index.ts"],"@xyo-network/plugins":["/Users/arietrouw/GitHub/XYOracleNetwork/sdk-xyo-client-js/src/index.ts"],"@xyo-network/sdk":["/Users/arietrouw/GitHub/XYOracleNetwork/sdk-xyo-client-js/src/index.ts"]},"invalidFiles":{},"invalidDirs":{}}

**Comment by arietrouw** [comment]
It looks like there was a pull request for a similar issue at one point: https://github.com/depcheck/depcheck/pull/725
